src/parser/expression.js

```diff
@@ -542,6 +542,10 @@ pp.parseMetaProperty = function (node, meta, propertyName) {
     this.raise(node.property.start, `The only valid meta property for new is ${meta.name}.${propertyName}`);
   }
 
+  if (!this.state.inFunction) {
+    this.raise(node.property.start, "new.target can only be used in functions");
+  }
```

Comment:
Could this be triggered also by `function.send`? If so I would rather do this check before/after calling this method.